tests/end-to-end/external-solution-integration/testkit/testservice.go

```diff
@@ -88,13 +89,17 @@ func (ts *testService) IsReady() (bool, error) {
 	url := ts.getHealthEndpointURL()
 	log.WithFields(log.Fields{"url": url}).Debug("IsReady?")
 
-	resp, _ := ts.HttpClient.Get(url)
+	resp, err := ts.HttpClient.Get(url)
 
-	// lets ignore all errors here, shall we?
+	if err != nil {
+		if strings.Contains(err.Error(), "connection reset by peer") {
+			log.WithFields(log.Fields{"err": err}).Debug("IsReady?")
```

Comment:
How is the `connection reset by peer` case different than the one below?

Reply:
`Connection reset by peer` is seen when the pod is still being brought up. So I ignore them for the time until its up. If there is other error message its handled